crankshaft-docker/src/service/builder.rs

```diff
@@ -173,6 +173,10 @@ impl Builder {
                             dir: self.work_dir,
                             env: Some(self.env.iter().map(|(k, v)| format!("{k}={v}")).collect()),
                             mounts: Some(self.mounts),
+                            // Ensure the caller's group id is added so that the container can
+                            // access the mounts and working directory
+                            #[cfg(unix)]
+                            groups: Some(vec![unsafe { libc::getegid() }.to_string()]),
```

Comment:
For the sake of keeping the `unsafe` count lower (at least directly in the crate), perhaps this is worth pulling in `nix`: https://docs.rs/nix/latest/nix/unistd/struct.Gid.html#method.effective